### ISSUE TITLE
[DEP0040] DeprecationWarning - punycode module is deprecated

### ISSUE THREAD

**Issue by philoserf** [DavidAnson/markdownlint#1032]
I used `NODE_OPTIONS='--trace-deprecation'` to get this stack trace:

```plain
[DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
    at node:punycode:3:9
    at BuiltinModule.compileForInternalLoader (node:internal/bootstrap/realm:392:7)
    at BuiltinModule.compileForPublicLoader (node:internal/bootstrap/realm:328:10)
    at loadBuiltinModule (node:internal/modules/helpers:101:7)
    at Module._load (node:internal/modules/cjs/loader:1001:17)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (/opt/homebrew/Cellar/markdownlint-cli/0.37.0/libexec/lib/node_modules/markdownlint-cli/node_modules/markdown-it/lib/index.js:14:20)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
```

It appears to be coming from markdown-it `node_modules/markdown-it/lib/index.js` via markdownlint.

see also: https://github.com/DavidAnson/markdownlint-cli2/issues/234 and https://github.com/igorshubovych/markdownlint-cli/issues/437

**Comment by thanakornkhi21** [comment]
> I used `NODE_OPTIONS='--trace-deprecation'` to get this stack trace:
> 
> ```
> [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
>     at node:punycode:3:9
>     at BuiltinModule.compileForInternalLoader (node:internal/bootstrap/realm:392:7)
>     at BuiltinModule.compileForPublicLoader (node:internal/bootstrap/realm:328:10)
>     at loadBuiltinModule (node:internal/modules/helpers:101:7)
>     at Module._load (node:internal/modules/cjs/loader:1001:17)
>     at Module.require (node:internal/modules/cjs/loader:1235:19)
>     at require (node:internal/modules/helpers:176:18)
>     at Object.<anonymous> (/opt/homebrew/Cellar/markdownlint-cli/0.37.0/libexec/lib/node_modules/markdownlint-cli/node_modules/markdown-it/lib/index.js:14:20)
>     at Module._compile (node:internal/modules/cjs/loader:1376:14)
>     at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
> ```
> 
> It appears to be coming from markdown-it `node_modules/markdown-it/lib/index.js` via markdownlint.
> 
> ดูเพิ่มเติมที่: [DavidAnson/markdownlint-cli2#234 ](https://github.com/DavidAnson/markdownlint-cli2/issues/234)and[igorshubovych/markdownlint-cli#437](https://github.com/igorshubovych/markdownlint-cli/issues/437)

**Comment by skyzyx** [comment]
Yep, been getting this too.

```bash
$ npx -y markdownlint-cli --fix '**/*.md' --ignore 'node_modules'
(node:85059) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.

(Use `node --trace-deprecation ...` to show where the warning was created)
```

```plain
$ node --version
v21.3.0

$ npm --version
10.2.4
```

**Comment by DavidAnson** [comment]
Once an updated version of `markdown-it` is available, I'll publish new releases.

**Comment by Callek** [comment]
@DavidAnson looks like `markdown-it` did just (a day or two ago) release their next version that includes this fix: https://github.com/markdown-it/markdown-it/commit/beed9aee2c1b44819afc25d485e86a6c699b6ef0

**Comment by DavidAnson** [comment]
Yes, that update has already been made to "next" branch: https://github.com/DavidAnson/markdownlint/commit/0e6ffaaab44cc3b229306866f04c991742587b4d

**Comment by Martiix** [comment]
When is next expected to be merged?

**Comment by DavidAnson** [comment]
I'll probably start a new round of releases in a week or two. This still isn't an issue for an LTS build of Node as far as I know.

https://github.com/DavidAnson/markdownlint/blob/main/doc/ReleaseProcess.md

**Comment by jarkenau** [comment]
Is there a workaround to use before the new release?

**Comment by DavidAnson** [comment]
Fixed in library version v0.33.0.
